Tasks/PowerShellOnTargetMachines/PowerShellJob.ps1

```diff
@@ -53,6 +53,55 @@ param (
     [String]$psOnRemoteScriptBlockString = "Invoke-PsOnRemote -MachineDnsName $fqdn -ScriptPath `$scriptPath -WinRMPort $port -Credential `$credential -ScriptArguments `$scriptArguments -InitializationScriptPath `$initializationScriptPath -SessionVariables `$parsedSessionVariables $skipCACheckOption $httpProtocolOption $enableDetailedLoggingOption"
     [scriptblock]$psOnRemoteScriptBlock = [scriptblock]::Create($psOnRemoteScriptBlockString)
     $deploymentResponse = Invoke-Command -ScriptBlock $psOnRemoteScriptBlock
-    
+   
+    # Telemetry data logic through ps session
+    try{
+        if($skipCACheckOption)
+        {
+            $sessionOption = New-PSSessionOption -SkipCACheck
+        }
+        else
+        {
+            $sessionOption = New-PSSessionOption 
+        }
+        $secpasswd = ConvertTo-SecureString  $credential.Password -AsPlainText -Force
+        $psCredential =  New-Object System.Management.Automation.PSCredential($credential.UserName, $secpasswd)
+        if($httpProtocolOption -eq '-UseHttp')
+        {
+            $session = New-PSSession -Computer $fqdn -Port $port -Credential $psCredential -SessionOption ($sessionOption )
+        }
+        else
+        {
+            $session = New-PSSession -Computer $fqdn -Port $port -Credential $psCredential -SessionOption ($sessionOption ) -UseSSL
+        }
+        $VmUuidHash  = Invoke-Command -Session  $session -ScriptBlock {
+            $sha = New-Object System.Security.Cryptography.SHA512CryptoServiceProvider
+            $computerDetails = Get-WmiObject -class Win32_ComputerSystemProduct -namespace root\CIMV2
+            $encoding = [system.Text.Encoding]::ASCII
+            $uuidHash = [System.BitConverter]::ToString( $sha.ComputeHash($encoding.GetBytes($computerDetails.UUID)))
+            $uuidHash = $uuidHash  -replace  "-"  , ""
+            return $uuidHash 
+        }
+        $isAzureVm = Invoke-Command -Session  $session -ScriptBlock {
+            (Get-Process -Name 'WindowsAzureGuestAgent' -ErrorAction Ignore) | Select-Object -First 1 |  ForEach-Object {
```

Comment:
do we put this in try catch on safer side

Reply:
no what I mean, do we need to put a try catch in ScriptBlock

Reply:
If script block will throw exception then it will be catched by outer try catch .